src/pybind/mgr/cephadm/upgrade.py

```diff
@@ -433,6 +433,31 @@ def _enough_mons_for_ok_to_stop(self) -> bool:
         mons = [m['name'] for m in j['monmap']['mons']]
         return len(mons) > 2
 
+    def _enough_mds_for_ok_to_stop(self, mds_daemon: DaemonDescription) -> bool:
+        # type (DaemonDescription) -> bool
+
+        # find fs this mds daemon belongs to
+        fsmap = self.mgr.get("fs_map")
+        for i in fsmap.get('filesystems', []):
+            fs = i["mdsmap"]
+            fs_name = fs["fs_name"]
+
+            assert mds_daemon.daemon_id
+            if fs_name != mds_daemon.service_name().split('.', 1)[1]:
+                # wrong fs for this mds daemon
+                continue
+
+            # get number of mds daemons for this fs
+            mds_count = len(
+                [daemon for daemon in self.mgr.cache.get_daemons_by_service(mds_daemon.service_name())])
+
+            # standby mds daemons for this fs?
+            if fs["max_mds"] < mds_count:
+                return True
+            return False
+
+        raise OrchestratorError(f'mds daemon {mds_daemon.name()} does not belong to any fs')
```

Comment:
This can happen, if you call `ceph orch apply mds fs_name` instead of `ceph fs volume create ...`. Would it make sense to succeed here?

Reply:
I just checked and it looks like mds daemons without a fs just pass ok-to-stop every time regardless of how many of them there are in the service so yes, we should be good to succeed here.